examples/ct/ct_handlers.go

```diff
@@ -321,13 +330,68 @@ func wrappedGetSTHConsistencyHandler(rpcClient trillian.TrillianLogClient) http.
 	}
 }
 
-func wrappedGetProofByHashHandler(rpcClient trillian.TrillianLogClient) http.HandlerFunc {
+func wrappedGetProofByHashHandler(c CTRequestHandlers) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		if !enforceMethod(w, r, httpMethodGet) {
 			return
 		}
 
-		http.Error(w, http.StatusText(http.StatusNotImplemented), http.StatusNotImplemented)
+		hash := r.FormValue(getProofParamHash)
+
+		// Accept any non empty hash that decodes from base64 and let the backend validate it further
+		if len(hash) == 0 {
+			sendHttpError(w, http.StatusBadRequest, errors.New("missing / empty hash param for get-proof-by-hash"))
+			return
+		}
+
+		leafHash, err := base64.StdEncoding.DecodeString(hash)
+
+		if err != nil {
+			sendHttpError(w, http.StatusBadRequest, errors.New("invalid base64 hash for get-proof-by-hash"))
+		}
+
+		rpcRequest := trillian.GetInclusionProofByHashRequest{LogId: c.logID, LeafHash: leafHash}
+		ctx, _ := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
+		response, err := c.rpcClient.GetInclusionProofByHash(ctx, &rpcRequest)
+
+		if err != nil || !rpcStatusOK(response.GetStatus()) {
+			sendHttpError(w, http.StatusInternalServerError, fmt.Errorf("RPC failed, possible extra info: %v", err))
+			return
+		}
+
+		// Per RFC 6962 section 4.5 we can't handle a case with duplicate leaves because the API is
+		// defined to return a single proof so it's an error if we don't get exactly one proof
```

Comment:
I'm also concerned that we could get a large number of results from storage in weird edge cases.